### PR TITLE
APP-348: Fix C++ Profiling demo

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @Slash32 @Dubland

### CODE REVIEW

**Comment by Dubland** [comment]
LGTM

File: scripts/apptester/apps.yam [hunk 1]
```diff
@@ -175,3 +175,18 @@ sparkdemo:
...
           buildcmd: "bash -c \"./build.sh build\""
+
+profilingdemo:
+  name: "Profiling demo"
+  language:
+    c:
+      src: "profilingdemo/source/c"
+      platform:
+        posix:
+          buildcmd: "bash -c \"./build.sh build\""
```
Comment (Slash32) [inline scripts/apptester/apps.yam:186]: dependency specification is missing

File: profilingdemo/source/cpp/CMakeLists.txt [hunk 2]
```diff
@@ -35,6 +35,6 @@ else (NOT DEFINED KAA_SDK_PATH)
 endif (NOT DEFINED KAA_SDK_PATH)
 
 add_executable(demo_client src/KaaDemo.cpp)
-target_link_libraries(demo_client kaacpp)
+target_link_libraries(demo_client kaacpp ${Boost_FILESYSTEM_LIBRARY})
```
Comment (Slash32) [inline profilingdemo/source/cpp/CMakeLists.txt:38]: Doesn't this need double-quotes?

**Review by Slash32 — CHANGES_REQUESTED** [review]
Minor nitpick, otherwise good.
Let's pass Jenkins.

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

File: scripts/apptester/apps.yam (hunk 1 above)
Comment (RostakaGmfun) [reply scripts/apptester/apps.yam:186]: There is no dependencies for C Profiling Demo, it is completely contained in `profilingdemo/source/c` directory

File: profilingdemo/source/cpp/CMakeLists.txt (hunk 2 above)
Comment (RostakaGmfun) [reply profilingdemo/source/cpp/CMakeLists.txt:38]: No, it doesn't. There might be a list of libraries in this variable and we want it to expand into a space-separated list of items

File: scripts/apptester/apps.yam (hunk 1 above)
Comment (Slash32) [reply scripts/apptester/apps.yam:186]: For other applications, there is dependency to build.sh script itself.

**Comment by Slash32** [comment]
The PR does not pass Jenkins. Please check https://jenkins.kaaproject.org:52001/job/KSAs_test_sample_apps_pr/23/consoleFull

**Comment by RostakaGmfun** [comment]
@kaa-leeroy test this fully please

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Comment by RostakaGmfun** [comment]
@kaa-leeroy test this fully please

**Review by Slash32 — APPROVED** [review]

**Pull request merged**